### ISSUE TITLE
Enum values in query parameters are not properly parsed

### ISSUE THREAD

**Issue by charles-crain** [grpc-ecosystem/grpc-gateway#366]
If you have an enum type in a gRPC request object, and you pass an enum value as a string as a query parameter, you get:
```
{
 "error": "strconv.ParseInt: parsing \"SOME_ENUM_VALUE\": invalid syntax",
 "code": 3
}
```

The generated Swagger passes the enum values as strings not integers, which leads to this error when using a Swagger generated client as well.

**Comment by tmc** [comment]
@charles-crain thanks for your report. Can you supply an example request for increased clarity here? Even better could be a failing test case.

https://github.com/grpc-ecosystem/grpc-gateway/pull/314 is related, FWIW.

**Comment by charles-crain** [comment]
Oh I think #314 is actually the same issue.  That issue shows it was fixed in release 1.2.0 correct?  I am using whatever the latest is from `go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway` ... does that lag behind?

An example request would look like this:

```
enum Field {
  FOO = 0;
  BAR = 1;
}

message FooRequest {
  Field which_field = 1;
}

service Foo {
  rpc FooCall (FooRequest) returns (FooResponse) {
    option (google.api.http) = {
            get: "/api/foo"
      };
  }
}

curl localhost/api/foo?which_field=BAR
```
